src/components/tree/tree.ts

```diff
@@ -0,0 +1,140 @@
+import { PluginObject } from 'vue';
+import Component from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+
+import { ModulVue } from '../../utils/vue/vue';
+import { TREE_NAME } from '../component-names';
+import I18nPlugin from '../i18n/i18n';
+import TreeNodePlugin from '../tree-node/tree-node';
+import WithRender from './tree.html?style=./tree.scss';
+
+export interface MTreeFormat {
+    idNode: string;
+    elementLabel?: string;
+    hasChildren?: boolean;
+}
+
+export interface TreeNode<T extends MTreeFormat> {
+    content: T;
+    children?: TreeNode<T>[];
+}
+
+export enum MSelectOption {
+    NONE = '0',
+    SINGLE = '1',
+    MULTIPLE = '2'
+}
+
+@WithRender
+@Component
+export class MTree<T extends MTreeFormat> extends ModulVue {
+
+    @Prop()
+    public tree: TreeNode<MTreeFormat>[];
+
+    @Prop({
+        default: MSelectOption.NONE,
+        validator: value =>
+            value === MSelectOption.NONE ||
+            value === MSelectOption.SINGLE ||
+            value === MSelectOption.MULTIPLE
+    })
+    public selectionQuantity: MSelectOption;
+
+    @Prop({ default: 'information' })
+    public selectionIcon: string;
+
+    @Prop()
+    public selectedNode: string[];
+
+    @Prop({ default: false })
+    public allOpen: boolean;
+
+    @Prop({ default: false })
+    public fileTree: boolean;
+
+    public emptyTreeTxt: string = this.$i18n.translate('m-tree:empty');
+    public errorTreeTxt: string = this.$i18n.translate('m-tree:error');
+    public treeVisibilityTxt: string = '';
+
+    private allOpenTxt: string = this.$i18n.translate('m-tree:all-close');
```

Comment:
au lieu de mettre un private member, je callerais direct this.$i18n dans setAllOpenTxt